src/v/cloud_storage/manifest.cc

```diff
@@ -223,6 +502,18 @@ void manifest::serialize(std::ostream& out) const {
             w.Int64(meta.committed_offset());
             w.Key("base_offset");
             w.Int64(meta.base_offset());
+            if (meta.base_timestamp != model::timestamp::missing()) {
```

Comment:
Nit: it would be nicer to have these attributes always present (but sometimes null) rather than having them be optional.  I'm thinking of it we ever have external tools reading them, the format is easier to describe if the list of attributes is always the same.